### PR TITLE
745: Check duplicate glossary entries

### LINKED ISSUE #745: Add validation to Glossary Terms creation for blocking duplicates
Hello,

Currently there's no validation on Glossary Term creation for blocking duplicates, would like to have that to avoid duplicates. Probably just block their creation and provide a warning/error message.

Screen showing duplicates;
screen shot 2017-06-23 at 10 10 31 am

Thank you

### PR DESCRIPTION
**Pull request by toolstack**
Resolves #745.

### CODE REVIEW

File: gp-includes/routes/glossary-entry.php [hunk 1]
```diff
@@ -81,15 +81,25 @@ public function glossary_entry_add_post( $project_path, $locale_slug, $translati
 			$this->redirect( gp_url_join( gp_url_project_locale( $project->path, $locale_slug, $translation_set_slug ), array( 'glossary' ) ) );
 		}
 		else {
-			$created_glossary_entry = GP::$glossary_entry->create_and_select( $new_glossary_entry );
+			$find_parms = array(
+				'glossary_id' => $glossary->id,
+				'term' => $new_glossary_entry->term,
```
Comment (ocean90) [inline gp-includes/routes/glossary-entry.php]: This should probably also include `$part_of_speech`

File: gp-includes/routes/glossary-entry.php [hunk 2]
```diff
@@ -81,15 +81,25 @@ public function glossary_entry_add_post( $project_path, $locale_slug, $translati
...
 
-			if ( ! $created_glossary_entry ) {
-				$this->errors[] = __( 'Error in creating glossary entry!', 'glotpress' );
-				$this->redirect( gp_url_join( gp_url_project_locale( $project->path, $locale_slug, $translation_set_slug ), array( 'glossary' ) ) );
-			}
-			else {
-				$this->notices[] = __( 'The glossary entry was created!', 'glotpress' );
+			if ( GP::$glossary_entry->find_one( $find_parms ) ) {
+				// Translators: %s is the glossary term that was attempted to be added.
+				$this->errors[] = sprintf( __( 'Error "%s" is a duplicate glossary entry!', 'glotpress' ), esc_html( $new_glossary_entry->term ) );
```
Comment (ocean90) [inline gp-includes/routes/glossary-entry.php]: Maybe make that message similar to https://github.com/GlotPress/GlotPress-WP/blob/cb79407aec395b4e3a5f6e0f9b3a79eb91914168/gp-includes/routes/translation.php#L290

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: gp-includes/routes/glossary-entry.php (hunk 1 above)
Comment (toolstack) [reply gp-includes/routes/glossary-entry.php]: Updated.

File: gp-includes/routes/glossary-entry.php (hunk 2 above)
Comment (toolstack) [reply gp-includes/routes/glossary-entry.php]: Updated.

**Review by ocean90 — APPROVED** [review]

**Pull request merged**